### ISSUE TITLE
move `process_ejections` to per-epoch

### ISSUE THREAD

**Issue by djrtwo** [ethereum/eth2.0-specs#305]
Ejections are currently in "per-block". They should instead be in per-epoch.

I argue per-epoch instead of per-slot because there is nothing during the middle of an epoch that lowers a validator's balance. Only after per-epoch rewards/penalties would a validator newly be below the `MIN_BALANCE` and qualify for ejection.

**Comment by JustinDrake** [comment]
Yep, makes sense 👍

**Comment by hwwhww** [comment]
closing in favor of #314
